plume-models/src/lists.rs

```diff
@@ -0,0 +1,588 @@
+    pub fn list_users(&self, conn: &Connection) -> Result<Vec<User>> {
+        list_elems::table
+            .filter(list_elems::list_id.eq(self.id))
+            .inner_join(users::table)
+            .select(users::all_columns)
+            .load(conn)
+            .map_err(Error::from)
+    }
+
+    /// Get all blogs in the list
+    pub fn list_blogs(&self, conn: &Connection) -> Result<Vec<Blog>> {
+        list_elems::table
+            .filter(list_elems::list_id.eq(self.id))
+            .inner_join(blogs::table)
+            .select(blogs::all_columns)
+            .load(conn)
+            .map_err(Error::from)
+    }
+
+    /// Get all words in the list
+    pub fn list_words(&self, conn: &Connection) -> Result<Vec<String>> {
+        if self.kind() != ListType::Word {
+            return Ok(vec![]);
+        }
+        list_elems::table
+            .filter(list_elems::list_id.eq(self.id))
+            .filter(list_elems::word.is_not_null())
+            .select(list_elems::word)
+            .load::<Option<String>>(conn)
+            .map_err(Error::from)
+            .map(|r| r.into_iter().filter_map(|o| o).collect::<Vec<String>>())
+    }
+
+    /// Get all prefixes in the list
+    pub fn list_prefixes(&self, conn: &Connection) -> Result<Vec<String>> {
+        if self.kind() != ListType::Prefix {
+            return Ok(vec![]);
+        }
+        list_elems::table
+            .filter(list_elems::list_id.eq(self.id))
+            .filter(list_elems::word.is_not_null())
+            .select(list_elems::word)
+            .load::<Option<String>>(conn)
+            .map_err(Error::from)
+            .map(|r| r.into_iter().filter_map(|o| o).collect::<Vec<String>>())
+    }
+
+    pub fn clear(&self, conn: &Connection) -> Result<()> {
+        diesel::delete(list_elems::table.filter(list_elems::list_id.eq(self.id)))
+            .execute(conn)
+            .map(|_| ())
+            .map_err(Error::from)
+    }
+
+    pub fn set_users(&self, conn: &Connection, users: &[i32]) -> Result<bool> {
+        if self.kind() != ListType::User {
+            return Ok(false);
+        }
+        self.clear(conn)?;
+        self.add_users(conn, users)
+    }
+
+    pub fn set_blogs(&self, conn: &Connection, blogs: &[i32]) -> Result<bool> {
+        if self.kind() != ListType::Blog {
+            return Ok(false);
+        }
+        self.clear(conn)?;
+        self.add_blogs(conn, blogs)
+    }
+
+    pub fn set_words(&self, conn: &Connection, words: &[&str]) -> Result<bool> {
+        if self.kind() != ListType::Word {
+            return Ok(false);
+        }
+        self.clear(conn)?;
+        self.add_words(conn, words)
+    }
+
+    pub fn set_prefixes(&self, conn: &Connection, prefixes: &[&str]) -> Result<bool> {
+        if self.kind() != ListType::Prefix {
+            return Ok(false);
+        }
+        self.clear(conn)?;
+        self.add_prefixes(conn, prefixes)
+    }
+}
+
+pub(super) mod private {
```

Comment:
this gives this module more visibility from the outer code, while keeping it private from other crates. It was probably necessary at some point, but rn it could be plain private